File: app/src/main/java/pt/estga/spotme/utils/PasswordUtils.java
package pt.estga.spotme.utils;

import org.mindrot.jbcrypt.BCrypt;

public class PasswordUtils {

    // Hashing da senha antes de armazenar na BD
    public static String hashPassword(String password) {
        return BCrypt.hashpw(password, BCrypt.gensalt());
    }

    // Verifica se a senha fornecida corresponde ao hash armazenado
    public static boolean verifyPassword(String password, String storedHash) {
        return BCrypt.checkpw(password, storedHash);
    }
}

File: app/src/main/java/pt/estga/spotme/utils/PasswordUtils.kt
package pt.estga.spotme.utils

import org.mindrot.jbcrypt.BCrypt

object PasswordUtils {
    // Hashing da senha antes de armazenar na BD
    fun hashPassword(password: String?): String {
        return BCrypt.hashpw(password, BCrypt.gensalt())
    }

    // Verifica se a senha fornecida corresponde ao hash armazenado
    fun verifyPassword(password: String?, storedHash: String?): Boolean {
        return BCrypt.checkpw(password, storedHash)
    }
}
